WordPress/gutenberg
# Mobile - Highlight words

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4175
- WordPress iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17381
- WordPress Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15504
- Aztec Android PR: https://github.com/wordpress-mobile/AztecEditor-Android/pull/943
- Aztec iOS PR: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1339

## Description
This PR adds the functionality to be able to customize colors for selected words. For now, users will only be able to change the text color, the functionality to customize the background color will not be supported on mobile for now.

<kbd><img src="https://user-images.githubusercontent.com/4885740/144085069-f5e53afa-a01e-4313-976f-1a7a97006bed.gif" width=200 /></kbd>

## How has this been tested?

## Test case 1 - Using a color from a theme's color palette

**Precondition**: Site with a universal theme e.g Geologist, Quadrat, Zoologist.

- Open the app and start a new post
- Type in some text
- Select a word or some words
- In the formating toolbar look for the `Highlight` option and select it
- **Expect** to see the bottom sheet open with the color palette of the theme
- Select one of the theme colors (not a custom one)
- Close the bottom sheet and expect to see the selected color applied
- Preview the post
- **Expect** to see the colors correctly
- Save the post and close it
- Open the created post
- **Expect** to see the color you added before

## Test case 2 - Using a custom color

**Precondition**: Site with a universal theme e.g Geologist, Quadrat, Zoologist.

- Open the app and start a new post
- Type in some text
- Select a word or some words
- In the formating toolbar look for the `Highlight` option and select it
- **Expect** to see the bottom sheet open with the color palette of the theme
- Select the `CUSTOM` option
- Choose any color and apply the changes
- Close the bottom sheet and expect to see the selected color applied
- Preview the post
- **Expect** to see the colors correctly
- Save the post and close it
- Open the created post
- **Expect** to see the color you added before

## Test case 3 - Using a color from the default color palette

**Precondition**: Site without any custom theme colors e.g Blank Canvas

- Open the app and start a new post
- Type in some text
- Select a word or some words
- In the formating toolbar look for the `Highlight` option and select it
- **Expect** to see the bottom sheet open with the **default** color palette
- Select one of the colors from the palette
- Close the bottom sheet and expect to see the selected color applied
- Preview the post
- **Expect** to see the colors correctly
- Save the post and close it
- Open the created post
- **Expect** to see the color you added before

## Screenshots 

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/140071269-d0fecd28-6d8d-4c9d-89b4-febaf8161784.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/140071278-f1938ce9-2168-431d-9910-bf9a6e8eff11.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/140071287-3da65aa8-c10a-4ed7-8baa-5acad6a5bfd1.gif" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
